### ISSUE TITLE
sys-module-configfs.device has increased boot time

### ISSUE THREAD

**Issue by MR-Diamond** [systemd/systemd#23850]
### systemd version the issue has been seen with

251

### Used distribution

Manjaro 21.3.1 Ruah

### Linux kernel version used

5.10.124-1-MANJARO

### CPU architectures issue was seen on

x86_64

### Component

systemd

### Expected behaviour you didn't see

I reported the same in another ticket: https://github.com/systemd/systemd/issues/23629#issuecomment-1153250281

The issue has started to appears since I updated to systemd 251;
with systemd 250 my system used to boot quite fast:

```
Startup finished in 1.123s (kernel) + 595ms (initrd) + 558ms (userspace) = 2.277s 
graphical.target reached after 558ms in userspace
```

Now with systemd 251:

```
Startup finished in 1.125s (kernel) + 584ms (initrd) + 1.509s (userspace) = 3.219s 
graphical.target reached after 1.506s in userspace.
```

Output of `systemd-analyze blame`:

https://pastebin.com/Luw1MCM5

The culprit seems to be `sys-module-configfs.device`

Please ask if more info are needed.

**Comment by yuwata** [comment]
Instead of systemd-analyze blame, please show the results of systemd-analyze plot for both v250 and v251.

**Comment by MR-Diamond** [comment]
v250 plot:
![2 277](https://user-images.githubusercontent.com/32249442/176176106-4a5bde2a-f7e3-45d2-9278-c0a558b9490e.svg)


v251 plot:
![3 278](https://user-images.githubusercontent.com/32249442/176176173-711aa6cc-5b6e-4dba-a801-5f30ee36d02c.svg)

**Comment by yuwata** [comment]
sys-module-configfs.device is activated too late, and thus modprobe@configfs.service is delayed. That seems to make many other late services are delayed to be started.

**Comment by yuwata** [comment]
I guess this is caused by 2336bde96420475ccb054326f27290fa0228f27d. Now block and tpm devices are processed with higher priority. But, fundamental modules like /sys/module/configfs shoud be processed much earlier, I guess.

**Comment by yuwata** [comment]
Fix is waiting in #23877.

@MR-Diamond Could you try the following?
```
$ sudo systemctl edit systemd-udev-trigger.service
[Service]
ExecStart=
ExecStart=-udevadm trigger --type=all --action=add --prioritized-subsystem=module,block,tpmrm,tty,net
```
Note that `udevadm trigger --type=all` is supported since v251. When you downgrade systemd to v250 or older, then you need to disable the modification, otherwise, the system may not boot correctly.

**Comment by MR-Diamond** [comment]
I edited the unit as you indicated, but this didn't helped:

```
Startup finished in 1.170s (kernel) + 663ms (initrd) + 977ms (userspace) = 2.811s 
graphical.target reached after 977ms in userspace.
```

plot:
![output](https://user-images.githubusercontent.com/32249442/176875213-19198d0e-f763-4324-b322-7ecde8ab7ba3.svg)

**Comment by yuwata** [comment]
@MR-Diamond Thanks.

About userspace,
v250: 558ms,
v251: 1.504s,
now: 977ms or 1.180s,

I think actually #23877 improves the situation.

Now the reason that, even with the above change, the boot process takes much time compared with v250 is that whole block device units must be re-processed by systemd-udevd before local-fs.target being ready. That's unfortunately expected result of recent change by 75d7b5989f99125e52d5c0e5656fa1cd0fae2405 and its follow-ups.

**Comment by MR-Diamond** [comment]
Why this ticket has been closed? https://github.com/systemd/systemd/pull/23877 doesn't really solve the performance regression.

**Comment by yuwata** [comment]
Interesting, could you provide a plot for the case?

**Comment by MR-Diamond** [comment]
> could you provide a plot for the case?

Sure:

```
Startup finished in 1.333s (kernel) + 627ms (initrd) + 1.783s (userspace) = 3.745s 
graphical.target reached after 1.541s in userspace.
```

Plot:
![output](https://user-images.githubusercontent.com/32249442/176960856-65c2d51a-f26d-4e17-a669-f9aab5184b19.svg)

**Comment by yuwata** [comment]
@MR-Diamond What is home-dave-mounts-servicedisk.mount?? Could you show the result of `systemctl status` and `systemctl list-dependencies --after` for the mount unit? Also, please provide the relevant fstab entry if exists.

**Comment by MR-Diamond** [comment]
> What is home-dave-mounts-servicedisk.mount??

My second SSD: `UUID=8412b438-5d7b-48bd-bfb1-46666bb7ac64	/home/dave/mounts/servicedisk/	ext4	defaults,noatime,commit=60	0	0`

```
systemctl status home-dave-mounts-servicedisk.mount
● home-dave-mounts-servicedisk.mount - /home/dave/mounts/servicedisk
     Loaded: loaded (/etc/fstab; generated)
     Active: active (mounted) since Fri 2022-07-01 21:53:58 CEST; 26min ago
      Until: Fri 2022-07-01 21:53:58 CEST; 26min ago
      Where: /home/dave/mounts/servicedisk
       What: /dev/sdb1
       Docs: man:fstab(5)
             man:systemd-fstab-generator(8)
      Tasks: 0 (limit: 19014)
     Memory: 96.0K
     CGroup: /system.slice/home-dave-mounts-servicedisk.mount

```

```
systemctl list-dependencies --after home-dave-mounts-servicedisk.mount
home-dave-mounts-servicedisk.mount
● ├─-.mount
● ├─dev-sdb1.device
● ├─system.slice
○ ├─systemd-journald.socket
○ ├─blockdev@dev-disk-by\x2duuid-8412b438\x2d5d7b\x2d48bd\x2dbfb1\x2d46666bb7ac64.target
○ ├─blockdev@dev-sdb1.target
● └─local-fs-pre.target
○   ├─dm-event.service
●   ├─systemd-remount-fs.service
●   └─systemd-tmpfiles-setup-dev.service

```

**Comment by yuwata** [comment]
With the change https://github.com/systemd/systemd/issues/23850#issuecomment-1172033846, actually sys-module-configfs.device is processed earlier.

But the boot time regression is caused by the home-dave-mounts-servicedisk.mount. I have no idea why it is started randomly delayed, sometimes soon after /dev/sdb1 is active (see https://github.com/systemd/systemd/issues/23850#issuecomment-1172185397), but sometimes ~0.7sec later after /dev/sdb1 is active (see https://github.com/systemd/systemd/issues/23850#issuecomment-1168646316 and https://github.com/systemd/systemd/issues/23850#issuecomment-1172670515)...

**Comment by yuwata** [comment]
@MR-Diamond
Let me confirm that the /dev/sdb1 is a plain ext4 partition, right?
If home-dave-mounts-servicedisk.mount is not critical to boot up the system, then could you try to comment out the fstab entry?
Could you show the results of `udevadm info /dev/sdb` and `udevadm info /dev/sdb1`?

**Comment by MR-Diamond** [comment]
> Let me confirm that the /dev/sdb1 is a plain ext4 partition, right?

Right, is ext4.

> If home-dave-mounts-servicedisk.mount is not critical to boot up the system, then could you try to comment out the fstab entry?

I comment it in fstab and now the boot time has been decreased al lot:

```
Startup finished in 1.154s (kernel) + 601ms (initrd) + 661ms (userspace) = 2.418s 
graphical.target reached after 661ms in userspace.
```
> Could you show the results of udevadm info /dev/sdb and udevadm info /dev/sdb1

`udevadm info /dev/sdb` :

```
P: /devices/pci0000:00/0000:00:1f.2/ata2/host1/target1:0:0/1:0:0:0/block/sdb
M: sdb
U: block
T: disk
D: b 8:16
N: sdb
L: 0
S: disk/by-path/pci-0000:00:1f.2-ata-2
S: disk/by-id/ata-CT250MX500SSD1_1801E109F5E0
S: disk/by-id/wwn-0x500a0751e109f5e0
S: disk/by-path/pci-0000:00:1f.2-ata-2.0
E: DEVPATH=/devices/pci0000:00/0000:00:1f.2/ata2/host1/target1:0:0/1:0:0:0/block/sdb
E: DEVNAME=/dev/sdb
E: DEVTYPE=disk
E: MAJOR=8
E: MINOR=16
E: SUBSYSTEM=block
E: USEC_INITIALIZED=2299326
E: ID_ATA=1
E: ID_TYPE=disk
E: ID_BUS=ata
E: ID_MODEL=CT250MX500SSD1
E: ID_MODEL_ENC=CT250MX500SSD1\x20\x20\x20\x20\x20\x20\x20\x20\x20\x20\x20\x20\x20\x20\x20\x20\x20\x20\x20\x20\x20\x20\x20\x20\x20\x20
E: ID_REVISION=M3CR023
E: ID_SERIAL=CT250MX500SSD1_1801E109F5E0
E: ID_SERIAL_SHORT=1801E109F5E0
E: ID_ATA_WRITE_CACHE=1
E: ID_ATA_WRITE_CACHE_ENABLED=1
E: ID_ATA_FEATURE_SET_PM=1
E: ID_ATA_FEATURE_SET_PM_ENABLED=1
E: ID_ATA_FEATURE_SET_SECURITY=1
E: ID_ATA_FEATURE_SET_SECURITY_ENABLED=0
E: ID_ATA_FEATURE_SET_SECURITY_ERASE_UNIT_MIN=2
E: ID_ATA_FEATURE_SET_SECURITY_ENHANCED_ERASE_UNIT_MIN=2
E: ID_ATA_FEATURE_SET_SECURITY_FROZEN=1
E: ID_ATA_FEATURE_SET_SMART=1
E: ID_ATA_FEATURE_SET_SMART_ENABLED=1
E: ID_ATA_FEATURE_SET_APM=1
E: ID_ATA_FEATURE_SET_APM_ENABLED=0
E: ID_ATA_DOWNLOAD_MICROCODE=1
E: ID_ATA_SATA=1
E: ID_ATA_SATA_SIGNAL_RATE_GEN2=1
E: ID_ATA_SATA_SIGNAL_RATE_GEN1=1
E: ID_ATA_ROTATION_RATE_RPM=0
E: ID_WWN=0x500a0751e109f5e0
E: ID_WWN_WITH_EXTENSION=0x500a0751e109f5e0
E: ID_PATH=pci-0000:00:1f.2-ata-2.0
E: ID_PATH_TAG=pci-0000_00_1f_2-ata-2_0
E: ID_PATH_ATA_COMPAT=pci-0000:00:1f.2-ata-2
E: ID_PART_TABLE_UUID=6a9bdb86
E: ID_PART_TABLE_TYPE=dos
E: DEVLINKS=/dev/disk/by-path/pci-0000:00:1f.2-ata-2 /dev/disk/by-id/ata-CT250MX500SSD1_1801E109F5E0 /dev/disk/by-id/wwn-0x500a0751e109f5e0 /dev/disk/by-path/pci-0000:00:1f.2-ata-2.0
E: TAGS=:systemd:
E: CURRENT_TAGS=:systemd:
```

`udevadm info /dev/sdb1`

```
P: /devices/pci0000:00/0000:00:1f.2/ata2/host1/target1:0:0/1:0:0:0/block/sdb/sdb1
M: sdb1
R: 1
U: block
T: partition
D: b 8:17
N: sdb1
L: 0
S: disk/by-label/servicedisk
S: disk/by-path/pci-0000:00:1f.2-ata-2.0-part1
S: disk/by-id/wwn-0x500a0751e109f5e0-part1
S: disk/by-path/pci-0000:00:1f.2-ata-2-part1
S: disk/by-partuuid/6a9bdb86-01
S: disk/by-id/ata-CT250MX500SSD1_1801E109F5E0-part1
S: disk/by-uuid/8412b438-5d7b-48bd-bfb1-46666bb7ac64
E: DEVPATH=/devices/pci0000:00/0000:00:1f.2/ata2/host1/target1:0:0/1:0:0:0/block/sdb/sdb1
E: DEVNAME=/dev/sdb1
E: DEVTYPE=partition
E: PARTN=1
E: MAJOR=8
E: MINOR=17
E: SUBSYSTEM=block
E: USEC_INITIALIZED=13256063
E: ID_ATA=1
E: ID_TYPE=disk
E: ID_BUS=ata
E: ID_MODEL=CT250MX500SSD1
E: ID_MODEL_ENC=CT250MX500SSD1\x20\x20\x20\x20\x20\x20\x20\x20\x20\x20\x20\x20\x20\x20\x20\x20\x20\x20\x20\x20\x20\x20\x20\x20\x20\x20
E: ID_REVISION=M3CR023
E: ID_SERIAL=CT250MX500SSD1_1801E109F5E0
E: ID_SERIAL_SHORT=1801E109F5E0
E: ID_ATA_WRITE_CACHE=1
E: ID_ATA_WRITE_CACHE_ENABLED=1
E: ID_ATA_FEATURE_SET_PM=1
E: ID_ATA_FEATURE_SET_PM_ENABLED=1
E: ID_ATA_FEATURE_SET_SECURITY=1
E: ID_ATA_FEATURE_SET_SECURITY_ENABLED=0
E: ID_ATA_FEATURE_SET_SECURITY_ERASE_UNIT_MIN=2
E: ID_ATA_FEATURE_SET_SECURITY_ENHANCED_ERASE_UNIT_MIN=2
E: ID_ATA_FEATURE_SET_SECURITY_FROZEN=1
E: ID_ATA_FEATURE_SET_SMART=1
E: ID_ATA_FEATURE_SET_SMART_ENABLED=1
E: ID_ATA_FEATURE_SET_APM=1
E: ID_ATA_FEATURE_SET_APM_ENABLED=0
E: ID_ATA_DOWNLOAD_MICROCODE=1
E: ID_ATA_SATA=1
E: ID_ATA_SATA_SIGNAL_RATE_GEN2=1
E: ID_ATA_SATA_SIGNAL_RATE_GEN1=1
E: ID_ATA_ROTATION_RATE_RPM=0
E: ID_WWN=0x500a0751e109f5e0
E: ID_WWN_WITH_EXTENSION=0x500a0751e109f5e0
E: ID_PATH=pci-0000:00:1f.2-ata-2.0
E: ID_PATH_TAG=pci-0000_00_1f_2-ata-2_0
E: ID_PATH_ATA_COMPAT=pci-0000:00:1f.2-ata-2
E: ID_PART_TABLE_UUID=6a9bdb86
E: ID_PART_TABLE_TYPE=dos
E: ID_FS_LABEL=servicedisk
E: ID_FS_LABEL_ENC=servicedisk
E: ID_FS_UUID=8412b438-5d7b-48bd-bfb1-46666bb7ac64
E: ID_FS_UUID_ENC=8412b438-5d7b-48bd-bfb1-46666bb7ac64
E: ID_FS_VERSION=1.0
E: ID_FS_TYPE=ext4
E: ID_FS_USAGE=filesystem
E: ID_PART_ENTRY_SCHEME=dos
E: ID_PART_ENTRY_UUID=6a9bdb86-01
E: ID_PART_ENTRY_TYPE=0x83
E: ID_PART_ENTRY_NUMBER=1
E: ID_PART_ENTRY_OFFSET=2048
E: ID_PART_ENTRY_SIZE=488394752
E: ID_PART_ENTRY_DISK=8:16
E: DEVLINKS=/dev/disk/by-label/servicedisk /dev/disk/by-path/pci-0000:00:1f.2-ata-2.0-part1 /dev/disk/by-id/wwn-0x500a0751e109f5e0-part1 /dev/disk/by-path/pci-0000:00:1f.2-ata-2-part1 /dev/disk/by-partuuid/6a9bdb86-01 /dev/disk/by-id/ata-CT250MX500SSD1_1801E109F5E0-part1 /dev/disk/by-uuid/8412b438-5d7b-48bd-bfb1-46666bb7ac64
E: TAGS=:systemd:
E: CURRENT_TAGS=:systemd:
```

Hower: I have this disk since thefebruary 2019 and it didn't gave me any issue, just only since systemd v251.

**Comment by MR-Diamond** [comment]
I discovered a worse thing. First of all: I rarely shutdown the laptop; I put it on sleep (suspend to ram) or I reboot it after system updates. Today I shutdown it (several times) and on cold boot it takes more than ten seconds! I attach the plot. I repeat: this occurs only on cold boot after a shutdown.
```

Startup finished in 1.321s (kernel) + 10.447s (initrd) + 659ms (userspace) = 12.428s 
graphical.target reached after 658ms in userspace
```

![output](https://user-images.githubusercontent.com/32249442/192092660-29ccb420-1f3b-4456-8258-108d8f3c46bb.svg)

**Comment by MR-Diamond** [comment]
>I discovered a worse thing [...] this occurs only on cold boot after a shutdown

 I forget to mention that I excluded the second SSD in `/etc/fstab`, by adding the parameter "noauto"; in this way, **after a reboot**, instead, I have a very fast startup:
 
```
Startup finished in 1.168s (kernel) + 625ms (initrd) + 649ms (userspace) = 2.443s 
graphical.target reached after 649ms in userspace.
```

And I can reconfirm that on a **cold boot after a shutdown** I always face 12 seconds startup.

**Comment by MR-Diamond** [comment]
> With the old systemd 250 I didn't had this issue on cold boot after shutdown.

Today, to be sure, I restored a TimeShift backup from May 2022 (which has systemd v 250); and in facts the looooong startup issue didn't occurred.

**Comment by MR-Diamond** [comment]
I also noticed another issue: after a cold boot, the usb wifi adapter is not recognized: I have to reboot to have it recognized. This doesn't occurs with systemd 250.

**Comment by yuwata** [comment]
In what sense, the adapter is not recognized? Please show the result of `udevadm info` for the WiFi interface, and shows relevant udevd debugging logs.

**Comment by MR-Diamond** [comment]
Is not recognized in the sense like if is unplugged: unplug and replug doesn't work: as I've said Iave to reboot it to have it active.

> relevant udevd debugging logs

What commands I have to provide?

**Comment by MR-Diamond** [comment]
> udevadm info

Results in "A device name or path is required"; Idk what path give.

**Comment by MR-Diamond** [comment]
Very strange... I shutdown and done a cold boot two times: this time the issue didn't occurred.

**Comment by MR-Diamond** [comment]
I have a (partially) good news. Today I received a system update and the loooong boot time after a cold boot is gone, but the slowdown with the second SSD is still there.
I summarize again:

Without second SSD:
![withoutsecondSSD](https://user-images.githubusercontent.com/32249442/199455496-91c46bae-2ef1-48dc-8860-20348c06e242.svg)

With second SSD:
![withsecondSSD](https://user-images.githubusercontent.com/32249442/199455562-b8f0f5d0-a4f3-45d0-bb63-f3fd3271d5f0.svg)

**Comment by yuwata** [comment]
@MR-Diamond Could you provide the version of systemd that now you use?

**Comment by MR-Diamond** [comment]
> Could you provide the version of systemd that now you use?
251.7-1

**Comment by yuwata** [comment]
Thanks. And, if you can remember that, could you tell us the exact previous systemd version when you reported? I noticed that you said v251, but we are not sure which point release you used.

**Comment by MR-Diamond** [comment]
251.2-3

**Comment by yuwata** [comment]
I guess, the additional fsck service for sdb1 is introduced by 13556724379a52951eb1977c2b7989a0159fd77c, which is backported as 73db7d99323c236625656f906eb4e429613d324b in v251.6.

Even if fsck is disabled in fstab, gpt-auto-generator may enable it.

@MR-Diamond If you really want to disable fsck for sdb1, then could you try to mask the systemd-fsck@.service: `sudo systemctl mask systemd-fsck@.service`? Then, please provide the plot in that case.

**Comment by MR-Diamond** [comment]
251.2-3

> I guess, the additional fsck service for sdb1 is introduced by [1355672](https://github.com/systemd/systemd/commit/13556724379a52951eb1977c2b7989a0159fd77c), which is backported as [73db7d9](https://github.com/systemd/systemd/commit/73db7d99323c236625656f906eb4e429613d324b) in v251.6.
> 
> Even if fsck is disabled in fstab, gpt-auto-generator may enable it.
> 
> @MR-Diamond If you really want to disable fsck for sdb1, then could you try to mask the [systemd-fsck@.service](mailto:systemd-fsck@.service): `sudo systemctl mask systemd-fsck@.service`? Then, please provide the plot in that case.

Yes: masking systemd-fsck@.service did the trick. The boot time in this case (with second SSD  enabled) was of 2.462s: (idk why but the plot always reports a wrong time).
I guess that I have to re-enable systemd-fsck@.service for security reasons?

![output](https://user-images.githubusercontent.com/32249442/199678678-6b57dfad-36bc-488a-af8d-55fb5e9cf64e.svg)

**Comment by yuwata** [comment]
> I guess that I have to re-enable `systemd-fsck@.service` for security reasons?

The fsck is not related to security, but for safety of your data in general. It is recommended that it is enabled.

But in this case, you explicitly disabled fsck in fstab, but that is not honored. I think it is a bug, though it is not serious, except for the boot time performance as you mentioned.

Could you provide the results of the following command?
- `systemctl cat systemd-fsck@dev-disk-by...service`
- `systemctl show systemd-fsck@dev-disk-by...service`

**Comment by MR-Diamond** [comment]
> The fsck is not related to security, but for safety

Yes, I meant safety, sorry.



> Could you provide the results of the following command?
> 
>     * `systemctl cat systemd-fsck@dev-disk-by...service`
> 
>     * `systemctl show systemd-fsck@dev-disk-by...service`

```
systemctl cat systemd-fsck@dev-disk-by...service
# /usr/lib/systemd/system/systemd-fsck@.service
#  SPDX-License-Identifier: LGPL-2.1-or-later
#
#  This file is part of systemd.
#
#  systemd is free software; you can redistribute it and/or modify it
#  under the terms of the GNU Lesser General Public License as published by
#  the Free Software Foundation; either version 2.1 of the License, or
#  (at your option) any later version.

[Unit]
Description=File System Check on %f
Documentation=man:systemd-fsck@.service(8)
DefaultDependencies=no
BindsTo=%i.device
Conflicts=shutdown.target
After=%i.device systemd-fsck-root.service local-fs-pre.target
Before=systemd-quotacheck.service shutdown.target

[Service]
Type=oneshot
RemainAfterExit=yes
ExecStart=/usr/lib/systemd/systemd-fsck %f
TimeoutSec=0
```

```
systemctl show systemd-fsck@dev-disk-by...service
Type=oneshot
ExitType=main
Restart=no
NotifyAccess=none
RestartUSec=100ms
TimeoutStartUSec=infinity
TimeoutStopUSec=infinity
TimeoutAbortUSec=infinity
TimeoutStartFailureMode=terminate
TimeoutStopFailureMode=terminate
RuntimeMaxUSec=infinity
RuntimeRandomizedExtraUSec=0
WatchdogUSec=infinity
WatchdogTimestampMonotonic=0
RootDirectoryStartOnly=no
RemainAfterExit=yes
GuessMainPID=yes
MainPID=0
ControlPID=0
FileDescriptorStoreMax=0
NFileDescriptorStore=0
StatusErrno=0
Result=success
ReloadResult=success
CleanResult=success
UID=[not set]
GID=[not set]
NRestarts=0
OOMPolicy=stop
ExecMainStartTimestampMonotonic=0
ExecMainExitTimestampMonotonic=0
ExecMainPID=0
ExecMainCode=0
ExecMainStatus=0
ExecStart={ path=/usr/lib/systemd/systemd-fsck ; argv[]=/usr/lib/systemd/systemd-fsck /dev/disk/by.. ; ignore_errors=no>
ExecStartEx={ path=/usr/lib/systemd/systemd-fsck ; argv[]=/usr/lib/systemd/systemd-fsck /dev/disk/by.. ; flags= ; start>
Slice=system-systemd\x2dfsck.slice
ControlGroupId=0
MemoryCurrent=[not set]
MemoryAvailable=infinity
CPUUsageNSec=[not set]
TasksCurrent=[not set]
IPIngressBytes=[no data]
IPIngressPackets=[no data]
IPEgressBytes=[no data]
IPEgressPackets=[no data]
IOReadBytes=18446744073709551615
IOReadOperations=18446744073709551615
IOWriteBytes=18446744073709551615
IOWriteOperations=18446744073709551615
Delegate=no
CPUAccounting=no
CPUWeight=[not set]
StartupCPUWeight=[not set]
CPUShares=[not set]
StartupCPUShares=[not set]
CPUQuotaPerSecUSec=infinity
CPUQuotaPeriodUSec=infinity
IOAccounting=no
IOWeight=[not set]
StartupIOWeight=[not set]
BlockIOAccounting=no
BlockIOWeight=[not set]
StartupBlockIOWeight=[not set]
MemoryAccounting=yes
DefaultMemoryLow=0
DefaultMemoryMin=0
MemoryMin=0
MemoryLow=0
MemoryHigh=infinity
MemoryMax=infinity
MemorySwapMax=infinity
MemoryLimit=infinity
DevicePolicy=auto
TasksAccounting=yes
TasksMax=19015
IPAccounting=no
ManagedOOMSwap=auto
ManagedOOMMemoryPressure=auto
ManagedOOMMemoryPressureLimit=0
ManagedOOMPreference=none
UMask=0022
LimitCPU=infinity
LimitCPUSoft=infinity
LimitFSIZE=infinity
LimitFSIZESoft=infinity
LimitDATA=infinity
LimitDATASoft=infinity
LimitSTACK=infinity
LimitSTACKSoft=8388608
LimitCORE=infinity
LimitCORESoft=infinity
LimitRSS=infinity
LimitRSSSoft=infinity
LimitNOFILE=524288
LimitNOFILESoft=1024
LimitAS=infinity
LimitASSoft=infinity
LimitNPROC=63383
LimitNPROCSoft=63383
LimitMEMLOCK=8388608
LimitMEMLOCKSoft=8388608
LimitLOCKS=infinity
LimitLOCKSSoft=infinity
LimitSIGPENDING=63383
LimitSIGPENDINGSoft=63383
LimitMSGQUEUE=819200
LimitMSGQUEUESoft=819200
LimitNICE=0
LimitNICESoft=0
LimitRTPRIO=0
LimitRTPRIOSoft=0
LimitRTTIME=infinity
LimitRTTIMESoft=infinity
OOMScoreAdjust=0
CoredumpFilter=0x33
Nice=0
IOSchedulingClass=2
IOSchedulingPriority=4
CPUSchedulingPolicy=0
CPUSchedulingPriority=0
CPUAffinityFromNUMA=no
NUMAPolicy=n/a
TimerSlackNSec=50000
CPUSchedulingResetOnFork=no
NonBlocking=no
StandardInput=null
StandardOutput=journal
StandardError=inherit
TTYReset=no
TTYVHangup=no
TTYVTDisallocate=no
SyslogPriority=30
SyslogLevelPrefix=yes
SyslogLevel=6
SyslogFacility=3
LogLevelMax=-1
LogRateLimitIntervalUSec=0
LogRateLimitBurst=0
SecureBits=0
CapabilityBoundingSet=cap_chown cap_dac_override cap_dac_read_search cap_fowner cap_fsetid cap_kill cap_setgid cap_setu>
DynamicUser=no
RemoveIPC=no
PrivateTmp=no
PrivateDevices=no
ProtectClock=no
ProtectKernelTunables=no
ProtectKernelModules=no
ProtectKernelLogs=no
ProtectControlGroups=no
PrivateNetwork=no
PrivateUsers=no
PrivateMounts=no
PrivateIPC=no
ProtectHome=no
ProtectSystem=no
SameProcessGroup=no
UtmpMode=init
IgnoreSIGPIPE=yes
NoNewPrivileges=no
SystemCallErrorNumber=2147483646
LockPersonality=no
RuntimeDirectoryPreserve=no
RuntimeDirectoryMode=0755
StateDirectoryMode=0755
CacheDirectoryMode=0755
LogsDirectoryMode=0755
ConfigurationDirectoryMode=0755
TimeoutCleanUSec=infinity
MemoryDenyWriteExecute=no
RestrictRealtime=no
RestrictSUIDSGID=no
RestrictNamespaces=no
MountAPIVFS=no
KeyringMode=private
ProtectProc=default
ProcSubset=all
ProtectHostname=no
KillMode=control-group
KillSignal=15
RestartKillSignal=15
FinalKillSignal=9
SendSIGKILL=yes
SendSIGHUP=no
WatchdogSignal=6
Id=systemd-fsck@dev-disk-by...service
Names=systemd-fsck@dev-disk-by...service
Requires="system-systemd\\x2dfsck.slice"
BindsTo=dev-disk-by...device
Conflicts=shutdown.target
Before=systemd-quotacheck.service shutdown.target
After=systemd-journald.socket systemd-fsck-root.service dev-disk-by...device local-fs-pre.target "system-systemd\\x2dfs>
Documentation="man:systemd-fsck@.service(8)"
Description=File System Check on /dev/disk/by..
LoadState=loaded
ActiveState=inactive
FreezerState=running
SubState=dead
FragmentPath=/usr/lib/systemd/system/systemd-fsck@.service
UnitFileState=static
UnitFilePreset=disabled
StateChangeTimestampMonotonic=0
InactiveExitTimestampMonotonic=0
ActiveEnterTimestampMonotonic=0
ActiveExitTimestampMonotonic=0
InactiveEnterTimestampMonotonic=0
CanStart=yes
CanStop=yes
CanReload=no
CanIsolate=no
CanFreeze=yes
StopWhenUnneeded=no
RefuseManualStart=no
RefuseManualStop=no
AllowIsolate=no
DefaultDependencies=no
OnSuccessJobMode=fail
OnFailureJobMode=replace
IgnoreOnIsolate=no
NeedDaemonReload=no
JobTimeoutUSec=infinity
JobRunningTimeoutUSec=infinity
JobTimeoutAction=none
ConditionResult=no
AssertResult=no
ConditionTimestampMonotonic=0
AssertTimestampMonotonic=0
Transient=no
Perpetual=no
StartLimitIntervalUSec=10s
StartLimitBurst=5
StartLimitAction=none
FailureAction=none
SuccessAction=none
CollectMode=inactive
```

**Comment by MR-Diamond** [comment]
> you explicitly disabled fsck in fstab

Where? This is the line, in fstab, for the second SSD (at the moment I just added `noauto`, for avoid the slowdown in the boot):

`UUID=8412b438-5d7b-48bd-bfb1-46666bb7ac64	/home/dave/mounts/servicedisk/	ext4	defaults,noatime,noauto,commit=60	0	2`

**Comment by MR-Diamond** [comment]
> My second SSD: `UUID=8412b438-5d7b-48bd-bfb1-46666bb7ac64 /home/dave/mounts/servicedisk/ ext4 defaults,noatime,commit=60 0 0`

Ah, ok. This is the line where I omitted fsck.

**Comment by MR-Diamond** [comment]
Today I received and update which include systemd 252.3-1 (from 251.8-1), and the startup time is doubled (with the second SSD still disabled in ect/fstab, the average time was around 2.400 seconds); now is around 4 seconds.

Plot:
![output](https://user-images.githubusercontent.com/32249442/208883611-e02c0ed4-7ad8-4b97-ba99-356f251219ca.svg)

**Comment by morialo3** [comment]
Same issue with systemd 252.4, also 
```
❯systemd-analyze                               
Startup finished in 7.253s (firmware) + 396ms (loader) + 1.770s (kernel) + 1.863s (initrd) + 5.840s (userspace) = 17.124s 
graphical.target reached after 5.840s in userspace.
```
![sap](https://user-images.githubusercontent.com/97380836/208927235-1fa09d2b-940c-4423-ad8f-05b511fd1436.svg)

**Comment by Zesko** [comment]
Out of my curiosity, what features increased boot time at Systemd **v252** and **v251**,  Systemd **v250** does not have?

**Comment by Ant59** [comment]
I can also confirm the same issue with v252.

```
❯ systemctl --version
systemd 252 (252.4-2-arch)
```

```
❯ cat /etc/fstab
# Static information about the filesystems.
# See fstab(5) for details.

# <file system>					<dir>		<type>	<options>				<dump>	<fsck>
UUID=D2CC-FDB1					/boot		vfat	defaults,noauto,x-systemd.automount	0	2
UUID=6ae602d6-0bfa-4e2b-baaf-46eea63a1efb	/		ext4	defaults				0	1
UUID=424847e5-c8c0-45a6-909e-4d13fde66d92	/mnt/games	ext4	defaults,noauto,x-systemd.automount	0	2
/swapfile 					none		swap	defaults				0	0
```

![plot](https://user-images.githubusercontent.com/556055/210339297-ef9cdad7-a45e-43ab-a041-5fef9b38a224.svg)

**Comment by meisme-dev** [comment]
I have a similar problem om systemd 251 
![image](https://user-images.githubusercontent.com/49573906/214876191-33e9fc73-b1e5-4943-a1d7-0b40c93f4b46.png)

**Comment by NikitaMGrimm** [comment]
Same problem here.
```
❯ uname -r                 
6.3.1-arch1-1
```

```
❯ systemctl --version                                  
systemd 253 (253.4-1-arch)
+PAM +AUDIT -SELINUX -APPARMOR -IMA +SMACK +SECCOMP +GCRYPT +GNUTLS +OPENSSL +ACL +BLKID +CURL +ELFUTILS +FIDO2 +IDN2 -IDN +IPTC +KMOD +LIBCRYPTSETUP +LIBFDISK +PCRE2 -PWQUALITY +P11KIT -QRENCODE +TPM2 +BZIP2 +LZ4 +XZ +ZLIB +ZSTD +BPF_FRAMEWORK +XKBCOMMON +UTMP -SYSVINIT default-hierarchy=unified
```
```
❯ cat /etc/fstab                     
# Static information about the filesystems.
# See fstab(5) for details.

# <file system> <dir> <type> <options> <dump> <pass>
# /dev/nvme0n1p4 LABEL=archroot
UUID=9d936343-59dd-4cb6-89e1-ee0b56eaf6a5	/         	btrfs     	rw,relatime,ssd,discard=async,space_cache=v2,subvolid=5,subvol=/	0 0

# /dev/nvme0n1p3
UUID=A7B6-C335      	/boot     	vfat      	rw,relatime,fmask=0022,dmask=0022,codepage=437,iocharset=ascii,shortname=mixed,utf8,noauto,x-systemd.automount,errors=remount-ro	0 2
```
```
❯ systemd-analyze                              
Startup finished in 8.304s (firmware) + 91ms (loader) + 526ms (kernel) + 5.147s (initrd) + 3.755s (userspace) = 17.825s 
graphical.target reached after 3.755s in userspace.
```
❯ systemd-analyze critical-chain           
The time when unit became active or started is printed after the "@>
The time the unit took to start is printed after the "+" character.

graphical.target @3.755s
└─multi-user.target @3.755s
  └─asusd.service @1.655s +2.099s
    └─power-profiles-daemon.service @1.629s +22ms
      └─basic.target @1.627s
        └─dbus-broker.service @1.616s +9ms
          └─dbus.socket @1.574s
            └─sysinit.target @1.573s
              └─systemd-backlight@leds:asus::kbd_backlight.service >
                └─system-systemd\x2dbacklight.slice @1.097s
                  └─system.slice
                    └─-.slice
```

```
6.054s dev-ttyS0.device
6.054s sys-devices-platform-serial8250-tty-ttyS0.device
6.054s dev-ttyS11.device
6.054s sys-devices-platform-serial8250-tty-ttyS11.device
6.053s dev-ttyS12.device
6.053s sys-devices-platform-serial8250-tty-ttyS12.device
6.053s dev-ttyS14.device
6.053s sys-devices-platform-serial8250-tty-ttyS14.device
6.053s sys-devices-platform-serial8250-tty-ttyS10.device
6.053s dev-ttyS10.device
6.052s sys-devices-platform-serial8250-tty-ttyS1.device
6.052s dev-ttyS1.device
6.052s dev-ttyS15.device
6.052s sys-devices-platform-serial8250-tty-ttyS15.device
6.052s dev-ttyS13.device
6.052s sys-devices-platform-serial8250-tty-ttyS13.device
6.051s dev-ttyS17.device
6.051s sys-devices-platform-serial8250-tty-ttyS17.device
6.051s sys-devices-platform-serial8250-tty-ttyS18.device
6.051s dev-ttyS18.device
6.051s dev-ttyS20.device
6.051s sys-devices-platform-serial8250-tty-ttyS20.device
6.051s dev-ttyS19.device
6.051s sys-devices-platform-serial8250-tty-ttyS19.device
6.050s dev-ttyS21.device
6.050s sys-devices-platform-serial8250-tty-ttyS21.device
6.050s sys-devices-platform-serial8250-tty-ttyS16.device
6.050s dev-ttyS16.device
6.050s dev-ttyS2.device
6.050s sys-devices-platform-serial8250-tty-ttyS2.device
6.049s sys-devices-platform-serial8250-tty-ttyS24.device
6.049s dev-ttyS24.device
6.048s dev-ttyS22.device
6.048s sys-devices-platform-serial8250-tty-ttyS22.device
6.048s dev-ttyS23.device
6.048s sys-devices-platform-serial8250-tty-ttyS23.device
6.048s dev-ttyS26.device
6.048s sys-devices-platform-serial8250-tty-ttyS26.device
6.047s dev-ttyS27.device
6.047s sys-devices-platform-serial8250-tty-ttyS27.device
6.047s sys-devices-platform-serial8250-tty-ttyS28.device
6.047s dev-ttyS28.device
6.047s dev-ttyS25.device
6.047s sys-devices-platform-serial8250-tty-ttyS25.device
6.047s sys-devices-platform-serial8250-tty-ttyS29.device
6.047s dev-ttyS29.device
6.046s dev-ttyS30.device
6.046s sys-devices-platform-serial8250-tty-ttyS30.device
6.046s dev-ttyS31.device
6.046s sys-devices-platform-serial8250-tty-ttyS31.device
6.046s sys-devices-platform-serial8250-tty-ttyS5.device
6.046s dev-ttyS5.device
6.046s sys-devices-platform-serial8250-tty-ttyS3.device
6.046s dev-ttyS3.device
6.046s sys-devices-platform-serial8250-tty-ttyS4.device
6.046s dev-ttyS4.device
6.045s sys-devices-platform-serial8250-tty-ttyS8.device
6.045s dev-ttyS8.device
6.045s sys-devices-platform-serial8250-tty-ttyS6.device
6.045s dev-ttyS6.device
6.045s dev-ttyS9.device
6.045s sys-devices-platform-serial8250-tty-ttyS9.device
6.044s sys-devices-platform-serial8250-tty-ttyS7.device
6.044s dev-ttyS7.device
6.012s sys-module-configfs.device
6.005s dev-disk-by\x2dpartlabel-Basic\x5cx20data\x5cx20partition.device
6.005s sys-devices-pci0000:00-0000:00:02.3-0000:06:00.0-nvme-nvme0-nvme0n1-nvme0n1p2.device
6.005s dev-disk-by\x2did-nvme\x2deui.002538581191c169\x2dpart2.device
6.005s dev-disk-by\x2duuid-D8761A3B761A1AB8.device
6.005s dev-nvme0n1p2.device
6.005s dev-disk-by\x2dpartuuid-f22ced9c\x2daf8f\x2d401c\x2d989f\x2d087b1af1622b.device
6.005s dev-disk-by\x2did-nvme\x2dSamsung_SSD_970_EVO_Plus_2TB_S4J4NM0R814502X_1\x2dpart2.device
6.005s dev-disk-by\x2ddiskseq-1\x2dpart2.device
6.005s dev-disk-by\x2did-nvme\x2dSamsung_SSD_970_EVO_Plus_2TB_S4J4NM0R814502X\x2dpart2.device
6.005s dev-disk-by\x2dlabel-Secondary.device
6.005s dev-disk-by\x2dpath-pci\x2d0000:06:00.0\x2dnvme\x2d1\x2dpart2.device
6.005s dev-disk-by\x2dlabel-archroot.device
6.005s dev-disk-by\x2duuid-9d936343\x2d59dd\x2d4cb6\x2d89e1\x2dee0b56eaf6a5.device
6.005s dev-disk-by\x2dpath-pci\x2d0000:06:00.0\x2dnvme\x2d1\x2dpart4.device
6.005s dev-disk-by\x2ddiskseq-1\x2dpart4.device
6.005s dev-disk-by\x2did-nvme\x2dSamsung_SSD_970_EVO_Plus_2TB_S4J4NM0R814502X\x2dpart4.device
6.005s dev-disk-by\x2did-nvme\x2dSamsung_SSD_970_EVO_Plus_2TB_S4J4NM0R814502X_1\x2dpart4.device
6.005s dev-nvme0n1p4.device
6.005s dev-disk-by\x2dpartuuid-e19f0d31\x2da3bf\x2dd54b\x2d8c58\x2dfafc61f67b56.device
6.005s dev-disk-by\x2did-nvme\x2deui.002538581191c169\x2dpart4.device
6.005s sys-devices-pci0000:00-0000:00:02.3-0000:06:00.0-nvme-nvme0-nvme0n1-nvme0n1p4.device
6.005s dev-disk-by\x2did-nvme\x2dSamsung_SSD_970_EVO_Plus_2TB_S4J4NM0R814502X_1\x2dpart3.device
6.005s dev-disk-by\x2dpartuuid-1f3a5da4\x2df767\x2da040\x2da1ca\x2d82533a360e88.device
6.005s dev-disk-by\x2ddiskseq-1\x2dpart3.device
6.005s sys-devices-pci0000:00-0000:00:02.3-0000:06:00.0-nvme-nvme0-nvme0n1-nvme0n1p3.device
6.005s dev-disk-by\x2did-nvme\x2dSamsung_SSD_970_EVO_Plus_2TB_S4J4NM0R814502X\x2dpart3.device
6.005s dev-disk-by\x2dpath-pci\x2d0000:06:00.0\x2dnvme\x2d1\x2dpart3.device
6.005s dev-nvme0n1p3.device
6.005s dev-disk-by\x2duuid-A7B6\x2dC335.device
6.005s dev-disk-by\x2did-nvme\x2deui.002538581191c169\x2dpart3.device
6.005s dev-disk-by\x2did-nvme\x2dSamsung_SSD_970_EVO_Plus_2TB_S4J4NM0R814502X_1.device
6.005s dev-disk-by\x2did-nvme\x2deui.002538581191c169.device
6.005s dev-disk-by\x2ddiskseq-1.device
6.005s dev-disk-by\x2did-nvme\x2dSamsung_SSD_970_EVO_Plus_2TB_S4J4NM0R814502X.device
6.005s sys-devices-pci0000:00-0000:00:02.3-0000:06:00.0-nvme-nvme0-nvme0n1.device
6.005s dev-nvme0n1.device
6.005s dev-disk-by\x2dpath-pci\x2d0000:06:00.0\x2dnvme\x2d1.device
6.004s dev-disk-by\x2dpartuuid-cdc4ac74\x2dd980\x2d47bf\x2dba9f\x2dbd9f44a9d524.device
6.004s sys-devices-pci0000:00-0000:00:02.3-0000:06:00.0-nvme-nvme0-nvme0n1-nvme0n1p1.device
6.004s dev-disk-by\x2dpath-pci\x2d0000:06:00.0\x2dnvme\x2d1\x2dpart1.device
6.004s dev-disk-by\x2did-nvme\x2dSamsung_SSD_970_EVO_Plus_2TB_S4J4NM0R814502X\x2dpart1.device
6.004s dev-disk-by\x2ddiskseq-1\x2dpart1.device
6.004s dev-disk-by\x2dpartlabel-Microsoft\x5cx20reserved\x5cx20partition.device
6.004s dev-disk-by\x2did-nvme\x2deui.002538581191c169\x2dpart1.device
6.004s dev-nvme0n1p1.device
6.004s dev-disk-by\x2did-nvme\x2dSamsung_SSD_970_EVO_Plus_2TB_S4J4NM0R814502X_1\x2dpart1.device
5.941s sys-devices-pci0000:00-0000:00:02.4-0000:07:00.0-nvme-nvme1-nvme1n1.device
5.941s dev-disk-by\x2did-nvme\x2deui.002538d6114254bc.device
5.941s dev-disk-by\x2did-nvme\x2dSAMSUNG_MZVLQ1T0HBLB\x2d00B00_S6F7NJ0R603323_1.device
5.941s dev-disk-by\x2did-nvme\x2dSAMSUNG_MZVLQ1T0HBLB\x2d00B00_S6F7NJ0R603323.device
5.941s dev-disk-by\x2dpath-pci\x2d0000:07:00.0\x2dnvme\x2d1.device
5.941s dev-disk-by\x2ddiskseq-2.device
5.941s dev-nvme1n1.device
2.099s asusd.service
 387ms initrd-switch-root.service
 338ms systemd-udev-trigger.service
 315ms systemd-remount-fs.service
 282ms systemd-modules-load.service
 240ms user@1000.service
 189ms systemd-tmpfiles-setup-dev.service
 172ms modprobe@fuse.service
 145ms systemd-tmpfiles-setup.service
 140ms systemd-journal-flush.service
 133ms lvm2-monitor.service
 132ms modprobe@drm.service
 123ms systemd-backlight@backlight:amdgpu_bl1.service
 106ms systemd-random-seed.service
  84ms systemd-update-utmp.service
  78ms systemd-user-sessions.service
  63ms kmod-static-nodes.service
  60ms systemd-journald.service
  58ms user-runtime-dir@1000.service
  55ms systemd-sysctl.service
  39ms systemd-backlight@leds:asus::kbd_backlight.service
  37ms modprobe@dm_mod.service
  35ms systemd-rfkill.service
  31ms systemd-udevd.service
  31ms systemd-tmpfiles-clean.service
  29ms NetworkManager.service
  22ms power-profiles-daemon.service
  19ms systemd-binfmt.service
  17ms systemd-logind.service
  15ms bluetooth.service
  12ms sys-fs-fuse-connections.mount
  11ms sys-kernel-config.mount
   9ms dbus-broker.service
   9ms dev-hugepages.mount
   8ms polkit.service
   8ms dev-mqueue.mount
   7ms sys-kernel-debug.mount
   7ms sys-kernel-tracing.mount
   7ms tmp.mount
   6ms avahi-daemon.service
   5ms initrd-parse-etc.service
   5ms supergfxd.service
   4ms initrd-cleanup.service
   3ms initrd-udevadm-cleanup-db.service
   3ms wpa_supplicant.service
   2ms rtkit-daemon.service
   1ms modprobe@configfs.service
   1ms modprobe@loop.service
   1ms proc-sys-fs-binfmt_misc.mount
```
```
❯ cat /etc/mkinitcpio.conf
MODULES=(btrfs)
BINARIES=()
FILES=()
HOOKS=(systemd keyboard autodetect modconf kms block)
COMPRESSION="lz4"

MODULES_DECOMPRESS="yes"
```
```
efibootmgr --disk /dev/nvme0n1 --part 3 --create --label "Arch-Linux" --loader /vmlinuz-linux --unicode 'root=UUID=9d936343-59dd-4cb6-89e1-ee0b56eaf6a5 rw quiet loglevel=3 systemd.show_status=auto rd.udev.log_level=3 vt.global_cursor_default=0 quiet fsck.mode=skip loglevel=3 systemd.show_status=auto rd.udev.log_level=3 vt.global_cursor_default=0 rootflags=subvol=/ initrd=\amd-ucode.img initrd=\initramfs-linux.img'
```

**Comment by Zesko** [comment]
Try switching from `systemd` hook to `udev` hook in the file `/etc/mkinitcpio.conf`

**Comment by whoisnobody** [comment]
I experience the same issue. Any news on this?

**Comment by MR-Diamond** [comment]
With Systemd v. 256.9 the boot time is even worse:

https://github.com/systemd/systemd/pull/23877#issuecomment-2558552775

**Comment by hackcr1258** [comment]
I experience the same issue too.

**Comment by bitmural** [comment]
my systemd-analyze looks similar, with a weirdly long sys-module-configfs.device with 2 SSD's. Is there a way to troubleshoot it? Other than knowing the timings, not sure where else to check. I don't see anything interesting really in journalctl

**Comment by yuwata** [comment]
Maybe fixed by 0494ced5a9897dc1bdbdd142c61bf6762b2265bf ? Though, the commit mentions proc-sys-fs-binfmt_misc.mount, but not sys-module-configfs.device.

Could you test the commit?

Also, try to boot with `systemd.log_level=debug` kernel command line option, then find which unit is blocked by the .device unit, or which unit blocks the .device unit.
